app/Http/Controllers/ImagesController.php

```diff
@@ -65,26 +65,60 @@ public function show(Post $post, Request $request)
     /**
      * Edits and overwrites an image based on given request parameters.
      *
+     * @TODO - Currently rotates and overwrites both the original image and
+     * processed, edited image. We will not need to work with the edited image
+     * when Glide processing is turned on, so we should remove this logic when
+     * that is live on prod.
+     *
      * @param  $id
      * @param  Request $request
      * @return \Illuminate\Http\Response
      */
     public function update($id, Request $request)
     {
         $post = Post::findOrFail($id);
-        $originalImage = $post->getMediaUrl();
+
+        // Get the filename of the original image.
+        $originalImage = $post->url;
+        $originalFilename = $this->getFilenameFromUrl($originalImage);
+
+        // Get the url of the processed, edited image.
+        $editedImage = $post->getMediaUrl();
 
         // Only supports rotation, right now.
         if ($request->input('rotate')) {
             $value = (int) -$request->input('rotate');
 
             $originalImage = Image::make($originalImage)->rotate($value)->encode('jpg', 75);
```

Comment:
Let's not re-encode the original image as a JPEG, so that if we have a higher quality PNG we don't lose anything (and so that multiple rotations don't re-compress the image each time).